packages/react-dom/src/events/DOMTopLevelEventTypes.js

```diff
@@ -0,0 +1,195 @@
+export const TOP_EMPTIED: TopLevelType = 'emptied';
+export const TOP_ENCRYPTED: TopLevelType = 'encrypted';
+export const TOP_ENDED: TopLevelType = 'ended';
+export const TOP_ERROR: TopLevelType = 'error';
+export const TOP_FOCUS: TopLevelType = 'focus';
+export const TOP_INPUT: TopLevelType = 'input';
+export const TOP_INVALID: TopLevelType = 'invalid';
+export const TOP_KEY_DOWN: TopLevelType = 'keydown';
+export const TOP_KEY_PRESS: TopLevelType = 'keypress';
+export const TOP_KEY_UP: TopLevelType = 'keyup';
+export const TOP_LOAD: TopLevelType = 'load';
+export const TOP_LOAD_START: TopLevelType = 'loadstart';
+export const TOP_LOADED_DATA: TopLevelType = 'loadeddata';
+export const TOP_LOADED_METADATA: TopLevelType = 'loadedmetadata';
+export const TOP_MOUSE_DOWN: TopLevelType = 'mousedown';
+export const TOP_MOUSE_MOVE: TopLevelType = 'mousemove';
+export const TOP_MOUSE_OUT: TopLevelType = 'mouseout';
+export const TOP_MOUSE_OVER: TopLevelType = 'mouseover';
+export const TOP_MOUSE_UP: TopLevelType = 'mouseup';
+export const TOP_PASTE: TopLevelType = 'paste';
+export const TOP_PAUSE: TopLevelType = 'pause';
+export const TOP_PLAY: TopLevelType = 'play';
+export const TOP_PLAYING: TopLevelType = 'playing';
+export const TOP_PROGRESS: TopLevelType = 'progress';
+export const TOP_RATE_CHANGE: TopLevelType = 'ratechange';
+export const TOP_RESET: TopLevelType = 'reset';
+export const TOP_SCROLL: TopLevelType = 'scroll';
+export const TOP_SEEKED: TopLevelType = 'seeked';
+export const TOP_SEEKING: TopLevelType = 'seeking';
+export const TOP_SELECTION_CHANGE: TopLevelType = 'selectionchange';
+export const TOP_STALLED: TopLevelType = 'stalled';
+export const TOP_SUBMIT: TopLevelType = 'submit';
+export const TOP_SUSPEND: TopLevelType = 'suspend';
+export const TOP_TEXT_INPUT: TopLevelType = 'textInput';
+export const TOP_TIME_UPDATE: TopLevelType = 'timeupdate';
+export const TOP_TOGGLE: TopLevelType = 'toggle';
+export const TOP_TOUCH_CANCEL: TopLevelType = 'touchcancel';
+export const TOP_TOUCH_END: TopLevelType = 'touchend';
+export const TOP_TOUCH_MOVE: TopLevelType = 'touchmove';
+export const TOP_TOUCH_START: TopLevelType = 'touchstart';
+export const TOP_TRANSITION_END: TopLevelType = getVendorPrefixedEventName(
+  'transitionend',
+);
+export const TOP_VOLUME_CHANGE: TopLevelType = 'volumechange';
+export const TOP_WAITING: TopLevelType = 'waiting';
+export const TOP_WHEEL: TopLevelType = 'wheel';
+
+export const mediaEventTypes: Array<TopLevelType> = [
+  TOP_ABORT,
+  TOP_CAN_PLAY,
+  TOP_CAN_PLAY_THROUGH,
+  TOP_DURATION_CHANGE,
+  TOP_EMPTIED,
+  TOP_ENCRYPTED,
+  TOP_ENDED,
+  TOP_ERROR,
+  TOP_LOADED_DATA,
+  TOP_LOADED_METADATA,
+  TOP_LOAD_START,
+  TOP_PAUSE,
+  TOP_PLAY,
+  TOP_PLAYING,
+  TOP_PROGRESS,
+  TOP_RATE_CHANGE,
+  TOP_SEEKED,
+  TOP_SEEKING,
+  TOP_STALLED,
+  TOP_SUSPEND,
+  TOP_TIME_UPDATE,
+  TOP_VOLUME_CHANGE,
+  TOP_WAITING,
+];
+
+export function getRawEventName(topLevelType: TopLevelType): string {
```

Comment:
This might no be necessary but I figured it was a good idea especially since `DOMTopLevelEventType` are opaque and other modules don't know that they hold the raw event name and (as I verified) GCC eliminates this function.